### PR TITLE
[tensorflow-cc] on x64 Windows with dynamic linkage misses C++ symbols in DLL

### LINKED ISSUE #18887: [tensorflow] Remove tensorflow packet on Windows
Tensorflow on Windows is unusable, without users manually hacking the produced libraries.

https://github.com/tensorflow/tensorflow/issues/23542
https://stackoverflow.com/questions/62954634/tensorflow-2-3-unresolved-external-symbols-in-machine-generated-files-when-build

This is confusing since "ok" installed packets fail during link phase because missing exported symbols.

Regardless of "who should fix it/whose fault it is" fact is that having a broken package in vcpkg is bad for user perspective of vcpkg quality. 

As I presume that it is unlikely somebody will write a magic fix for this since it does not seem trivial so I think proper thing is to remove the tensorflow packet for Windows. 

If it matters broken version I tried is 2.4#2, but I presume all of them are broken.

note: build(installing tensorflow_cc with vcpkg) works fine, I get .lib and .dll (in both Release and Debug variants), but if you actually try to actually write some TF code that uses those libs then you get LNK2019.

as for repro:
install tensorflow_cc and try to actually use to build some toy TF project. Some of missing symbols are:

LNK2019: unresolved external symbol "public: __cdecl tensorflow::SessionOptions::SessionOptions(void)”

LNK2019: unresolved external symbol "class tensorflow::Status __cdecl tensorflow::NewSession(struct tensorflow::SessionOptions const &,class tensorflow::Session * *)"

### PR DESCRIPTION
**Pull request by jgehw**
DLL appears to only export C API, but not C++ API. Always build static C++ library until this is fixed upstream.

Fixes #18887.

### CODE REVIEW

File: ports/tensorflow-cc/portfile.cmake [hunk 1]
```diff
@@ -1,5 +1,11 @@
 vcpkg_fail_port_install(ON_ARCH "x86" "arm" ON_TARGET "UWP")
 
+if(VCPKG_TARGET_IS_WINDOWS)
+	if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
+		message(FATAL_ERROR "TensorFlow doesn't support the C++ API via dynamic linkage on Windows. Please consider using static linkage (x64-windows-static triplet).")
+	endif()
```
Comment (Hoikas) [inline ports/tensorflow-cc/portfile.cmake:6]:
```suggestion
    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -1756,3 +1756,6 @@ cppgraphqlgen:x64-osx=fail
 # Changes in Python have broken tensorflow on our osx hardware
 tensorflow:x64-osx=fail
 tensorflow-cc:x64-osx=fail
+
+# TensorFlow x64 Windows DLL misses C++ API symbols, only C API exported; needs to be fixed upstream
+tensorflow-cc:x64-windows=fail
```
Comment (Hoikas) [inline scripts/ci.baseline.txt:1761]:
```suggestion
```

`vcpkg_check_linkage()` will fix this.

**Comment by Hoikas** [comment]
> Issue fatal error until this is fixed upstream.

With the changes, the library now always builds static on Windows, so the PR description should probably be updated 😉

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
LGTM on my side.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**